aesara/compile/function/__init__.py

```diff
@@ -86,18 +92,18 @@ def function_dump(
 
 
 def function(
-    inputs,
-    outputs=None,
-    mode=None,
-    updates=None,
-    givens=None,
-    no_default_updates=False,
-    accept_inplace=False,
-    name=None,
-    rebuild_strict=True,
-    allow_input_downcast=None,
-    profile=None,
-    on_unused_input=None,
+    inputs: "List[Variable]",
+    outputs: "Optional[List[Variable]]" = None,
+    mode: Union[str] = None,
```

Comment:
```suggestion
    mode: Union[str, "Mode"] = None,
```
And further above in the `TYPE_CHECKING` block `from aesara.compile.mode import Mode`

Reply:
Shouldn't the type-hint be optional?